docs/src/components/ColorSwatch/ColorSwatch.component.tsx

```diff
@@ -0,0 +1,110 @@
+import { Pill } from '../Pill'
+import { CopyToClipboard } from 'react-copy-to-clipboard';
+import { Typography } from '../../../../src/Typography';
+// THEME
+import { colors } from '../../../../src/theme';
+
+
+// INTERFACES
+interface ColorSwatchProps {
+  colorHexCode: string;
+  colorName: string;
+}
+
+
+// STYLES
+const StyledColorSwatch = styled.div`
+  border: 0.5rem solid ${props => props.highContrastBorderColor}};
+  display: inline-block;
+  margin: 0 2rem 2rem 0;
+  width: 17rem;
+
+  .color-swatch-pill-container {
+    background: ${props => props.colorHexCode };
+    cursor: pointer;
+    padding: 7rem 1rem 1rem 1.5rem;
+  }
+
+  .color-swatch-text-container {
+    border-top: 0.05rem solid ${props => props.highContrastBorderColor};
+    padding: 1rem;
+  }
+
+  .color-swatch-text-container p {
+    margin: 0;
+    padding: 0.25rem 0 0.25rem 0.5rem;
+  }
+
+  .color-swatch-text-container p:hover {
+    background: ${colors.ash.light};
+    border-radius: 1rem;
+    cursor: pointer;
+    text-decoration: underline;
+  }
+
+  .temporary-copy-confirmation {
+    font-size: 1rem;
+    font-weight: 600;
+    position: fixed;
+    padding: 1rem;
+    top: 2rem;
+    right: 2rem;
+    z-index: 10;
+    color: ${colors.success.base};
+    background: ${colors.white.base};
+    border: 0.1rem solid ${colors.success.base};
+  }
+`;
+
+
+// JSX MARKUP
+export const ColorSwatch = ({
+  colorHexCode,
+  colorName
+}: ColorSwatchProps): React.ReactElement<any> => {
+
+  const highContrastBorderColor = ColorHelpers.getHigherContrastColor(colorHexCode);
+  const colorRgbCode = ColorHelpers.getRgbColor(colorHexCode)
+
+  const [isCopied, setIsCopied] = useState<boolean>(false);
+  const [copyToClipboardText, setCopyToClipboardText] = useState<string>(colorHexCode);
+  const handleCopy = (copiedText) => {
+    setCopyToClipboardText(copiedText)
+    setIsCopied(true);
+    setTimeout(() => {setIsCopied(false)}, 1000);
+  }
+
+  return (
+    <StyledColorSwatch
+      colorHexCode={colorHexCode}
+      highContrastBorderColor={highContrastBorderColor}
+    >
+
+    {isCopied ? <span className="temporary-copy-confirmation">🎉 You copied "{copyToClipboardText}"</span> : null}
+
+      <CopyToClipboard text={colorHexCode} onCopy={() => handleCopy(colorHexCode)}>
+        <div className="color-swatch-pill-container">
+          {['#FFFFFF', '#FAFAFA', '#F1F1F1', '#000000'].map((pillBackgroundColor: string) => (
+            <Pill
+              key={pillBackgroundColor}
+              value={ColorHelpers.getRoundedColorContrast(colorHexCode, `${pillBackgroundColor}`)}
+              backgroundColor={pillBackgroundColor}
+              textColor={colorHexCode}
+            />
+          ))}
+        </div>
+     </CopyToClipboard>
+     <div className="color-swatch-text-container">
+       <CopyToClipboard text={colorName} onCopy={() => handleCopy(colorName)}><Typography as="p" weight="700">{colorName}</Typography></CopyToClipboard>
```

Comment:
This is now
```
<Typography as="p" weight="700">
```
this
```
<Typography tag="p" weight="700">
```
because of a bug identified by @brandonlilly

Reply:
It looks like `Typography` was also updated to zero out margin and padding, which I was taking advantage of before. What is the preferred way to add in margin or padding?

Reply:
✅ I added this margin back in using the styled component.